src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -0,0 +1,726 @@
+        $adResourceId = $configDpInfo.adResourceId
+
+        # If the health check fails (not 200 response), an exception is thrown
+        # so we can ignore the output.
+        $null = Invoke-ConfigDPHealthCheck -configDPEndpoint $configDPEndpoint -Resource $adResourceId
+
+        # This call does the "pure ARM" update of the ARM objects.
+        Write-Debug "Writing Connected Kubernetes ARM objects."
+
+        # Re-put here
+        $Response = Az.ConnectedKubernetes.internal\New-AzConnectedKubernetes @PSBoundParameters
+
+        # Retrieving Helm chart OCI (Open Container Initiative) Artifact location
+        Write-Debug "Retrieving Helm chart OCI (Open Container Initiative) Artifact location."
+        Write-Debug "PUT response: $Response"
+        $ResponseStr = "$Response"
+        $helmValuesDp = Get-HelmValues `
+            -configDPEndpoint $configDPEndpoint `
+            -releaseTrain $ReleaseTrain `
+            -requestBody $ResponseStr `
+            -Verbose:($PSCmdlet.MyInvocation.BoundParameters["Verbose"].IsPresent -eq $true) `
+            -Debug:($PSCmdlet.MyInvocation.BoundParameters["Debug"].IsPresent -eq $true)
+
+        Write-Debug "helmValuesDp: $helmValuesDp"
+        Write-Debug "OCI Artifact location: ${helmValuesDp.repositoryPath}."
+
+        # Allow a custom OCI registry to be set via environment variables.
+        # !!PDS: Where are these variables documented?  Should they be?
+        #
+        # AZURE_ACCESS_TOKEN
+        # HELMCHART
+        # HELMREGISTRY
+        # HELMVALUESPATH
+        # RELEASETRAIN
+        # USERPROFILE
+        #
+        $registryPath = if ($env:HELMREGISTRY) { $env:HELMREGISTRY } else { $helmValuesDp.repositoryPath }
+        Write-Debug "RegistryPath: ${registryPath}."
+
+        $helmValuesContent = $helmValuesDp.helmValuesContent
+        Write-Debug "Helm values: ${helmValuesContent}."
+
+        foreach ($field in $helmValuesContent.PSObject.Properties) {
+            if ($field.Name -in @("global.httpsProxy", "global.httpProxy", "global.noProxy", "global.proxyCert")) {
+                continue
+            }
+            $options += " --set $($field.Name)=$($field.Value)"
+        }
+        
+        # Get helm chart path (within the OCI registry).
+        $chartPath = Get-HelmChartPath -registryPath $registryPath -kubeConfig $KubeConfig -kubeContext $KubeContext -helmClientLocation $HelmClientLocation
+        if (Test-Path Env:HELMCHART) {
+            $ChartPath = Get-ChildItem -Path Env:HELMCHART
+        }
+
+        $TenantId = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile.DefaultContext.Tenant.Id
+        try {
+            helm upgrade `
+            --debug `
+            azure-arc `
+            $ChartPath `
+            --namespace $ReleaseInstallNamespace `
+            --wait (-split $options)
+        } catch {
+            throw "Unable to install helm release at $ChartPath"
```

Comment:
"Unable to update helm release" perhaps?

Reply:
Sounds good to me! Updated.